dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/TimeoutFilter.java

```diff
@@ -36,21 +36,34 @@
 
     private static final Logger logger = LoggerFactory.getLogger(TimeoutFilter.class);
 
+    private static final String TIMEOUT_FILTER_START_TIME = "timeout_filter_start_time";
+
     @Override
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
-        long start = System.currentTimeMillis();
-        Result result = invoker.invoke(invocation);
-        long elapsed = System.currentTimeMillis() - start;
-        if (invoker.getUrl() != null
-                && elapsed > invoker.getUrl().getMethodParameter(invocation.getMethodName(),
-                "timeout", Integer.MAX_VALUE)) {
-            if (logger.isWarnEnabled()) {
-                logger.warn("invoke time out. method: " + invocation.getMethodName()
-                        + " arguments: " + Arrays.toString(invocation.getArguments()) + " , url is "
-                        + invoker.getUrl() + ", invoke elapsed " + elapsed + " ms.");
+        if (invocation.getAttachments() != null) {

```

Comment:
If there is no attachement for invocation, to handle fix timeout should we  care to create a attachment and set TIMEOUT_FILTER_START_TIME ? What do you say?

Reply:
Seems like you are right...
Great suggestion, I will fix it.
👍

Reply:
I have fix this place.
Seems like only RpcInvocation can set attachments.

Reply:
I was think to have the responsibility of assigning a default TIMEOUT_FILTER_START_TIME  in RpcInvocation, but we will be landed up mixing its Responsibility with TimerFilter.  

I am just thinking loud here (please bear with me 😄 ), should we create a FilterWrapper and can store these information here, it may give us better way of separating object information. e.g

```
Class FilterWrapper {
  private Invocation target;
  private Map<String,Object> properties = new HashMap<>();
   FilterWrapper (Invocation invocation)  {
      this.target = invocation;
  }
  public void addProperties(String key,Object value) {
   properties.put(key,value);
} 
   public Map<String,Object> getProperties() {
     return this.properties;
 }
   public Map<String,Object> getAttachment() {
      this.target.getAttachment();
  }
  ..........
_more getter and setter or deleter as per needs_
  ....
}
wrapper=FilterWrapper.of(invocation); 
wrapper.addProperties(TIMEOUT_FILTER_START_TIME, String.valueOf(start));
```

So even in future we need to keep add internal properties we may not be landed up overriding user propvided values.
What do you say?

Reply:
@khanimteyaz That's interesting, but if we add an extra `properties` just like you did in FilterWrapper, we need to further figure out how to pass the `properties` onto the wire. But at present `invocation` is the only carrier, so everything needs to finally go into `invocation` to get transferred.